// @aws-sdk/client-ivs-realtime/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "ivs",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class IVSRealTimeClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultIVSRealTimeHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

let IVSRealTimeServiceException$1 = class IVSRealTimeServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, IVSRealTimeServiceException.prototype);
    }
};

let AccessDeniedException$1 = class AccessDeniedException extends IVSRealTimeServiceException$1 {
    name = "AccessDeniedException";
    $fault = "client";
    accessControlAllowOrigin;
    accessControlExposeHeaders;
    cacheControl;
    contentSecurityPolicy;
    strictTransportSecurity;
    xContentTypeOptions;
    xFrameOptions;
    xAmznErrorType;
    exceptionMessage;
    constructor(opts) {
        super({
            name: "AccessDeniedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, AccessDeniedException.prototype);
        this.accessControlAllowOrigin = opts.accessControlAllowOrigin;
        this.accessControlExposeHeaders = opts.accessControlExposeHeaders;
        this.cacheControl = opts.cacheControl;
        this.contentSecurityPolicy = opts.contentSecurityPolicy;
        this.strictTransportSecurity = opts.strictTransportSecurity;
        this.xContentTypeOptions = opts.xContentTypeOptions;
        this.xFrameOptions = opts.xFrameOptions;
        this.xAmznErrorType = opts.xAmznErrorType;
        this.exceptionMessage = opts.exceptionMessage;
    }
};
let ConflictException$1 = class ConflictException extends IVSRealTimeServiceException$1 {
    name = "ConflictException";
    $fault = "client";
    accessControlAllowOrigin;
    accessControlExposeHeaders;
    cacheControl;
    contentSecurityPolicy;
    strictTransportSecurity;
    xContentTypeOptions;
    xFrameOptions;
    xAmznErrorType;
    exceptionMessage;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.accessControlAllowOrigin = opts.accessControlAllowOrigin;
        this.accessControlExposeHeaders = opts.accessControlExposeHeaders;
        this.cacheControl = opts.cacheControl;
        this.contentSecurityPolicy = opts.contentSecurityPolicy;
        this.strictTransportSecurity = opts.strictTransportSecurity;
        this.xContentTypeOptions = opts.xContentTypeOptions;
        this.xFrameOptions = opts.xFrameOptions;
        this.xAmznErrorType = opts.xAmznErrorType;
        this.exceptionMessage = opts.exceptionMessage;
    }
};
let InternalServerException$1 = class InternalServerException extends IVSRealTimeServiceException$1 {
    name = "InternalServerException";
    $fault = "server";
    accessControlAllowOrigin;
    accessControlExposeHeaders;
    cacheControl;
    contentSecurityPolicy;
    strictTransportSecurity;
    xContentTypeOptions;
    xFrameOptions;
    xAmznErrorType;
    exceptionMessage;
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.accessControlAllowOrigin = opts.accessControlAllowOrigin;
        this.accessControlExposeHeaders = opts.accessControlExposeHeaders;
        this.cacheControl = opts.cacheControl;
        this.contentSecurityPolicy = opts.contentSecurityPolicy;
        this.strictTransportSecurity = opts.strictTransportSecurity;
        this.xContentTypeOptions = opts.xContentTypeOptions;
        this.xFrameOptions = opts.xFrameOptions;
        this.xAmznErrorType = opts.xAmznErrorType;
        this.exceptionMessage = opts.exceptionMessage;
    }
};
let PendingVerification$1 = class PendingVerification extends IVSRealTimeServiceException$1 {
    name = "PendingVerification";
    $fault = "client";
    accessControlAllowOrigin;
    accessControlExposeHeaders;
    cacheControl;
    contentSecurityPolicy;
    strictTransportSecurity;
    xContentTypeOptions;
    xFrameOptions;
    xAmznErrorType;
    exceptionMessage;
    constructor(opts) {
        super({
            name: "PendingVerification",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PendingVerification.prototype);
        this.accessControlAllowOrigin = opts.accessControlAllowOrigin;
        this.accessControlExposeHeaders = opts.accessControlExposeHeaders;
        this.cacheControl = opts.cacheControl;
        this.contentSecurityPolicy = opts.contentSecurityPolicy;
        this.strictTransportSecurity = opts.strictTransportSecurity;
        this.xContentTypeOptions = opts.xContentTypeOptions;
        this.xFrameOptions = opts.xFrameOptions;
        this.xAmznErrorType = opts.xAmznErrorType;
        this.exceptionMessage = opts.exceptionMessage;
    }
};
let ResourceNotFoundException$1 = class ResourceNotFoundException extends IVSRealTimeServiceException$1 {
    name = "ResourceNotFoundException";
    $fault = "client";
    accessControlAllowOrigin;
    accessControlExposeHeaders;
    cacheControl;
    contentSecurityPolicy;
    strictTransportSecurity;
    xContentTypeOptions;
    xFrameOptions;
    xAmznErrorType;
    exceptionMessage;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.accessControlAllowOrigin = opts.accessControlAllowOrigin;
        this.accessControlExposeHeaders = opts.accessControlExposeHeaders;
        this.cacheControl = opts.cacheControl;
        this.contentSecurityPolicy = opts.contentSecurityPolicy;
        this.strictTransportSecurity = opts.strictTransportSecurity;
        this.xContentTypeOptions = opts.xContentTypeOptions;
        this.xFrameOptions = opts.xFrameOptions;
        this.xAmznErrorType = opts.xAmznErrorType;
        this.exceptionMessage = opts.exceptionMessage;
    }
};
let ServiceQuotaExceededException$1 = class ServiceQuotaExceededException extends IVSRealTimeServiceException$1 {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    accessControlAllowOrigin;
    accessControlExposeHeaders;
    cacheControl;
    contentSecurityPolicy;
    strictTransportSecurity;
    xContentTypeOptions;
    xFrameOptions;
    xAmznErrorType;
    exceptionMessage;
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
        this.accessControlAllowOrigin = opts.accessControlAllowOrigin;
        this.accessControlExposeHeaders = opts.accessControlExposeHeaders;
        this.cacheControl = opts.cacheControl;
        this.contentSecurityPolicy = opts.contentSecurityPolicy;
        this.strictTransportSecurity = opts.strictTransportSecurity;
        this.xContentTypeOptions = opts.xContentTypeOptions;
        this.xFrameOptions = opts.xFrameOptions;
        this.xAmznErrorType = opts.xAmznErrorType;
        this.exceptionMessage = opts.exceptionMessage;
    }
};
let ValidationException$1 = class ValidationException extends IVSRealTimeServiceException$1 {
    name = "ValidationException";
    $fault = "client";
    accessControlAllowOrigin;
    accessControlExposeHeaders;
    cacheControl;
    contentSecurityPolicy;
    strictTransportSecurity;
    xContentTypeOptions;
    xFrameOptions;
    xAmznErrorType;
    exceptionMessage;
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
        this.accessControlAllowOrigin = opts.accessControlAllowOrigin;
        this.accessControlExposeHeaders = opts.accessControlExposeHeaders;
        this.cacheControl = opts.cacheControl;
        this.contentSecurityPolicy = opts.contentSecurityPolicy;
        this.strictTransportSecurity = opts.strictTransportSecurity;
        this.xContentTypeOptions = opts.xContentTypeOptions;
        this.xFrameOptions = opts.xFrameOptions;
        this.xAmznErrorType = opts.xAmznErrorType;
        this.exceptionMessage = opts.exceptionMessage;
    }
};

const _ACAO = "Access-Control-Allow-Origin";
const _ACEH = "Access-Control-Expose-Headers";
const _ADE = "AccessDeniedException";
const _APRC = "AutoParticipantRecordingConfiguration";
const _C = "Composition";
const _CC = "Cache-Control";
const _CDC = "ChannelDestinationConfiguration";
const _CE = "ConflictException";
const _CEC = "CreateEncoderConfiguration";
const _CECR = "CreateEncoderConfigurationRequest";
const _CECRr = "CreateEncoderConfigurationResponse";
const _CIC = "CreateIngestConfiguration";
const _CICR = "CreateIngestConfigurationRequest";
const _CICRr = "CreateIngestConfigurationResponse";
const _CPT = "CreateParticipantToken";
const _CPTR = "CreateParticipantTokenRequest";
const _CPTRr = "CreateParticipantTokenResponse";
const _CRHC = "CompositionRecordingHlsConfiguration";
const _CS = "CompositionSummary";
const _CSC = "CreateStorageConfiguration";
const _CSCR = "CreateStorageConfigurationRequest";
const _CSCRr = "CreateStorageConfigurationResponse";
const _CSL = "CompositionSummaryList";
const _CSP = "Content-Security-Policy";
const _CSR = "CreateStageRequest";
const _CSRr = "CreateStageResponse";
const _CSr = "CreateStage";
const _CTC = "CompositionThumbnailConfiguration";
const _CTCL = "CompositionThumbnailConfigurationList";
const _D = "Destination";
const _DC = "DestinationConfiguration";
const _DCL = "DestinationConfigurationList";
const _DD = "DestinationDetail";
const _DEC = "DeleteEncoderConfiguration";
const _DECR = "DeleteEncoderConfigurationRequest";
const _DECRe = "DeleteEncoderConfigurationResponse";
const _DIC = "DeleteIngestConfiguration";
const _DICR = "DeleteIngestConfigurationRequest";
const _DICRe = "DeleteIngestConfigurationResponse";
const _DL = "DestinationList";
const _DP = "DisconnectParticipant";
const _DPK = "DeletePublicKey";
const _DPKR = "DeletePublicKeyRequest";
const _DPKRe = "DeletePublicKeyResponse";
const _DPR = "DisconnectParticipantRequest";
const _DPRi = "DisconnectParticipantResponse";
const _DS = "DestinationSummary";
const _DSC = "DeleteStorageConfiguration";
const _DSCR = "DeleteStorageConfigurationRequest";
const _DSCRe = "DeleteStorageConfigurationResponse";
const _DSL = "DestinationSummaryList";
const _DSR = "DeleteStageRequest";
const _DSRe = "DeleteStageResponse";
const _DSe = "DeleteStage";
const _E = "Event";
const _EC = "EncoderConfiguration";
const _ECS = "EncoderConfigurationSummary";
const _ECSL = "EncoderConfigurationSummaryList";
const _EL = "EventList";
const _GC = "GridConfiguration";
const _GCR = "GetCompositionRequest";
const _GCRe = "GetCompositionResponse";
const _GCe = "GetComposition";
const _GEC = "GetEncoderConfiguration";
const _GECR = "GetEncoderConfigurationRequest";
const _GECRe = "GetEncoderConfigurationResponse";
const _GIC = "GetIngestConfiguration";
const _GICR = "GetIngestConfigurationRequest";
const _GICRe = "GetIngestConfigurationResponse";
const _GP = "GetParticipant";
const _GPK = "GetPublicKey";
const _GPKR = "GetPublicKeyRequest";
const _GPKRe = "GetPublicKeyResponse";
const _GPR = "GetParticipantRequest";
const _GPRe = "GetParticipantResponse";
const _GS = "GetStage";
const _GSC = "GetStorageConfiguration";
const _GSCR = "GetStorageConfigurationRequest";
const _GSCRe = "GetStorageConfigurationResponse";
const _GSR = "GetStageRequest";
const _GSRe = "GetStageResponse";
const _GSS = "GetStageSession";
const _GSSR = "GetStageSessionRequest";
const _GSSRe = "GetStageSessionResponse";
const _IC = "IngestConfiguration";
const _ICL = "IngestConfigurationList";
const _ICS = "IngestConfigurationSummary";
const _IPK = "ImportPublicKey";
const _IPKR = "ImportPublicKeyRequest";
const _IPKRm = "ImportPublicKeyResponse";
const _ISE = "InternalServerException";
const _LC = "LayoutConfiguration";
const _LCR = "ListCompositionsRequest";
const _LCRi = "ListCompositionsResponse";
const _LCi = "ListCompositions";
const _LEC = "ListEncoderConfigurations";
const _LECR = "ListEncoderConfigurationsRequest";
const _LECRi = "ListEncoderConfigurationsResponse";
const _LIC = "ListIngestConfigurations";
const _LICR = "ListIngestConfigurationsRequest";
const _LICRi = "ListIngestConfigurationsResponse";
const _LP = "ListParticipants";
const _LPE = "ListParticipantEvents";
const _LPER = "ListParticipantEventsRequest";
const _LPERi = "ListParticipantEventsResponse";
const _LPK = "ListPublicKeys";
const _LPKR = "ListPublicKeysRequest";
const _LPKRi = "ListPublicKeysResponse";
const _LPR = "ListParticipantsRequest";
const _LPRR = "ListParticipantReplicasRequest";
const _LPRRi = "ListParticipantReplicasResponse";
const _LPRi = "ListParticipantsResponse";
const _LPRis = "ListParticipantReplicas";
const _LS = "ListStages";
const _LSC = "ListStorageConfigurations";
const _LSCR = "ListStorageConfigurationsRequest";
const _LSCRi = "ListStorageConfigurationsResponse";
const _LSR = "ListStagesRequest";
const _LSRi = "ListStagesResponse";
const _LSS = "ListStageSessions";
const _LSSR = "ListStageSessionsRequest";
const _LSSRi = "ListStageSessionsResponse";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _P = "Participant";
const _PC = "PipConfiguration";
const _PK = "PublicKey";
const _PKL = "PublicKeyList";
const _PKS = "PublicKeySummary";
const _PL = "ParticipantList";
const _PR = "ParticipantReplica";
const _PRHC = "ParticipantRecordingHlsConfiguration";
const _PRL = "ParticipantReplicaList";
const _PS = "ParticipantSummary";
const _PT = "ParticipantToken";
const _PTC = "ParticipantThumbnailConfiguration";
const _PTCa = "ParticipantTokenConfiguration";
const _PTCar = "ParticipantTokenConfigurations";
const _PTL = "ParticipantTokenList";
const _PTS = "ParticipantTokenString";
const _PV = "PendingVerification";
const _RC = "RecordingConfiguration";
const _RNFE = "ResourceNotFoundException";
const _S = "Stage";
const _SC = "StorageConfiguration";
const _SCR = "StartCompositionRequest";
const _SCRt = "StartCompositionResponse";
const _SCRto = "StopCompositionRequest";
const _SCRtop = "StopCompositionResponse";
const _SCS = "StorageConfigurationSummary";
const _SCSL = "StorageConfigurationSummaryList";
const _SCt = "StartComposition";
const _SCto = "StopComposition";
const _SD = "S3Detail";
const _SDC = "S3DestinationConfiguration";
const _SE = "StageEndpoints";
const _SK = "StreamKey";
const _SPR = "StartParticipantReplication";
const _SPRR = "StartParticipantReplicationRequest";
const _SPRRt = "StartParticipantReplicationResponse";
const _SPRRto = "StopParticipantReplicationRequest";
const _SPRRtop = "StopParticipantReplicationResponse";
const _SPRt = "StopParticipantReplication";
const _SQEE = "ServiceQuotaExceededException";
const _SS = "StageSession";
const _SSC = "S3StorageConfiguration";
const _SSL = "StageSessionList";
const _SSLt = "StageSummaryList";
const _SSS = "StageSessionSummary";
const _SSt = "StageSummary";
const _STS = "Strict-Transport-Security";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TRRa = "TagResourceResponse";
const _UIC = "UpdateIngestConfiguration";
const _UICR = "UpdateIngestConfigurationRequest";
const _UICRp = "UpdateIngestConfigurationResponse";
const _UR = "UntagResource";
const _URR = "UntagResourceRequest";
const _URRn = "UntagResourceResponse";
const _US = "UpdateStage";
const _USR = "UpdateStageRequest";
const _USRp = "UpdateStageResponse";
const _V = "Video";
const _VE = "ValidationException";
const _XCTO = "X-Content-Type-Options";
const _XFO = "X-Frame-Options";
const _a = "arn";
const _aCAO = "accessControlAllowOrigin";
const _aCEH = "accessControlExposeHeaders";
const _aPRC = "autoParticipantRecordingConfiguration";
const _aSI = "activeSessionId";
const _at = "attributes";
const _b = "bitrate";
const _bN = "browserName";
const _bNu = "bucketName";
const _bV = "browserVersion";
const _c = "client";
const _cA = "channelArn";
const _cC = "cacheControl";
const _cSP = "contentSecurityPolicy";
const _ca = "capabilities";
const _ch = "channel";
const _co = "configuration";
const _com = "composition";
const _comp = "compositions";
const _d = "destinations";
const _dSA = "destinationStageArn";
const _dSI = "destinationSessionId";
const _de = "detail";
const _du = "duration";
const _e = "error";
const _eC = "encoderConfiguration";
const _eCA = "encoderConfigurationArn";
const _eCAn = "encoderConfigurationArns";
const _eCn = "encoderConfigurations";
const _eCr = "errorCode";
const _eM = "exceptionMessage";
const _eT = "endTime";
const _eTv = "eventTime";
const _eTx = "expirationTime";
const _en = "endpoints";
const _ev = "events";
const _f = "force";
const _fBECA = "filterByEncoderConfigurationArn";
const _fBP = "filterByPublished";
const _fBRS = "filterByRecordingState";
const _fBS = "filterByState";
const _fBSA = "filterByStageArn";
const _fBUI = "filterByUserId";
const _fJT = "firstJoinTime";
const _fPA = "featuredParticipantAttribute";
const _fi = "fingerprint";
const _fo = "format";
const _fr = "framerate";
const _g = "grid";
const _gG = "gridGap";
const _h = "height";
const _hC = "hlsConfiguration";
const _hE = "httpError";
const _hH = "httpHeader";
const _hQ = "httpQuery";
const _ht = "http";
const _i = "id";
const _iC = "ingestConfiguration";
const _iCn = "ingestConfigurations";
const _iI = "insecureIngest";
const _iN = "ispName";
const _iP = "ingestProtocol";
const _iT = "idempotencyToken";
const _l = "layout";
const _mR = "maxResults";
const _mT = "mediaTypes";
const _n = "name";
const _nT = "nextToken";
const _oN = "osName";
const _oSV = "omitStoppedVideo";
const _oV = "osVersion";
const _p = "participant";
const _pB = "pipBehavior";
const _pH = "pipHeight";
const _pI = "participantId";
const _pK = "publicKey";
const _pKM = "publicKeyMaterial";
const _pKu = "publicKeys";
const _pO = "pipOffset";
const _pOA = "participantOrderAttribute";
const _pP = "pipPosition";
const _pPA = "pipParticipantAttribute";
const _pT = "participantToken";
const _pTC = "participantTokenConfigurations";
const _pTa = "participantTokens";
const _pW = "pipWidth";
const _pa = "participants";
const _pi = "pip";
const _pr = "protocol";
const _pu = "published";
const _r = "reason";
const _rA = "resourceArn";
const _rC = "recordingConfiguration";
const _rM = "recordingMode";
const _rP = "recordingPrefix";
const _rPI = "remoteParticipantId";
const _rPR = "recordParticipantReplicas";
const _rRWS = "recordingReconnectWindowSeconds";
const _rS = "recordingState";
const _rSBN = "recordingS3BucketName";
const _rSP = "recordingS3Prefix";
const _rSe = "replicationState";
const _rT = "replicationType";
const _rWS = "reconnectWindowSeconds";
const _re = "replica";
const _rep = "replicas";
const _rt = "rtmp";
const _rtm = "rtmps";
const _s = "state";
const _sA = "stageArn";
const _sC = "storageConfiguration";
const _sCA = "storageConfigurationArn";
const _sCt = "storageConfigurations";
const _sI = "sessionId";
const _sK = "streamKey";
const _sS = "stageSession";
const _sSA = "sourceStageArn";
const _sSI = "sourceSessionId";
const _sSt = "stageSessions";
const _sT = "startTime";
const _sTS = "strictTransportSecurity";
const _sV = "sdkVersion";
const _s_ = "s3";
const _se = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.ivsrealtime";
const _st = "storage";
const _sta = "stage";
const _stag = "stages";
const _t = "tags";
const _tC = "thumbnailConfiguration";
const _tCh = "thumbnailConfigurations";
const _tIS = "targetIntervalSeconds";
const _tK = "tagKeys";
const _tSDS = "targetSegmentDurationSeconds";
const _to = "token";
const _uI = "userId";
const _v = "video";
const _vAR = "videoAspectRatio";
const _vFM = "videoFillMode";
const _w = "whip";
const _wi = "width";
const _xAET = "xAmznErrorType";
const _xCTO = "xContentTypeOptions";
const _xFO = "xFrameOptions";
const _xaE = "x-amzn-ErrorType";
const n0 = "com.amazonaws.ivsrealtime";
var ParticipantTokenString = [0, n0, _PTS, 8, 0];
var StreamKey = [0, n0, _SK, 8, 0];
var AccessDeniedException = [
    -3,
    n0,
    _ADE,
    {
        [_e]: _c,
        [_hE]: 403,
    },
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO, _xAET, _eM],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
        [
            0,
            {
                [_hH]: _xaE,
            },
        ],
        0,
    ],
];
schema.TypeRegistry.for(n0).registerError(AccessDeniedException, AccessDeniedException$1);
var AutoParticipantRecordingConfiguration = [
    3,
    n0,
    _APRC,
    0,
    [_sCA, _mT, _tC, _rRWS, _hC, _rPR],
    [0, 64 | 0, () => ParticipantThumbnailConfiguration, 1, () => ParticipantRecordingHlsConfiguration, 2],
];
var ChannelDestinationConfiguration = [3, n0, _CDC, 0, [_cA, _eCA], [0, 0]];
var Composition = [
    3,
    n0,
    _C,
    0,
    [_a, _sA, _s, _l, _d, _t, _sT, _eT],
    [0, 0, 0, () => LayoutConfiguration, () => DestinationList, 128 | 0, 5, 5],
];
var CompositionRecordingHlsConfiguration = [3, n0, _CRHC, 0, [_tSDS], [1]];
var CompositionSummary = [
    3,
    n0,
    _CS,
    0,
    [_a, _sA, _d, _s, _t, _sT, _eT],
    [0, 0, () => DestinationSummaryList, 0, 128 | 0, 5, 5],
];
var CompositionThumbnailConfiguration = [3, n0, _CTC, 0, [_tIS, _st], [1, 64 | 0]];
var ConflictException = [
    -3,
    n0,
    _CE,
    {
        [_e]: _c,
        [_hE]: 409,
    },
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO, _xAET, _eM],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
        [
            0,
            {
                [_hH]: _xaE,
            },
        ],
        0,
    ],
];
schema.TypeRegistry.for(n0).registerError(ConflictException, ConflictException$1);
var CreateEncoderConfigurationRequest = [
    3,
    n0,
    _CECR,
    0,
    [_n, _v, _t],
    [0, () => Video, 128 | 0],
];
var CreateEncoderConfigurationResponse = [
    3,
    n0,
    _CECRr,
    0,
    [_eC],
    [() => EncoderConfiguration],
];
var CreateIngestConfigurationRequest = [
    3,
    n0,
    _CICR,
    0,
    [_n, _sA, _uI, _at, _iP, _iI, _t],
    [0, 0, 0, 128 | 0, 0, 2, 128 | 0],
];
var CreateIngestConfigurationResponse = [
    3,
    n0,
    _CICRr,
    0,
    [_iC],
    [[() => IngestConfiguration, 0]],
];
var CreateParticipantTokenRequest = [
    3,
    n0,
    _CPTR,
    0,
    [_sA, _du, _uI, _at, _ca],
    [0, 1, 0, 128 | 0, 64 | 0],
];
var CreateParticipantTokenResponse = [
    3,
    n0,
    _CPTRr,
    0,
    [_pT],
    [[() => ParticipantToken, 0]],
];
var CreateStageRequest = [
    3,
    n0,
    _CSR,
    0,
    [_n, _pTC, _t, _aPRC],
    [0, () => ParticipantTokenConfigurations, 128 | 0, () => AutoParticipantRecordingConfiguration],
];
var CreateStageResponse = [
    3,
    n0,
    _CSRr,
    0,
    [_sta, _pTa],
    [() => Stage, [() => ParticipantTokenList, 0]],
];
var CreateStorageConfigurationRequest = [
    3,
    n0,
    _CSCR,
    0,
    [_n, _s_, _t],
    [0, () => S3StorageConfiguration, 128 | 0],
];
var CreateStorageConfigurationResponse = [
    3,
    n0,
    _CSCRr,
    0,
    [_sC],
    [() => StorageConfiguration],
];
var DeleteEncoderConfigurationRequest = [3, n0, _DECR, 0, [_a], [0]];
var DeleteEncoderConfigurationResponse = [3, n0, _DECRe, 0, [], []];
var DeleteIngestConfigurationRequest = [3, n0, _DICR, 0, [_a, _f], [0, 2]];
var DeleteIngestConfigurationResponse = [3, n0, _DICRe, 0, [], []];
var DeletePublicKeyRequest = [3, n0, _DPKR, 0, [_a], [0]];
var DeletePublicKeyResponse = [3, n0, _DPKRe, 0, [], []];
var DeleteStageRequest = [3, n0, _DSR, 0, [_a], [0]];
var DeleteStageResponse = [3, n0, _DSRe, 0, [], []];
var DeleteStorageConfigurationRequest = [3, n0, _DSCR, 0, [_a], [0]];
var DeleteStorageConfigurationResponse = [3, n0, _DSCRe, 0, [], []];
var Destination = [
    3,
    n0,
    _D,
    0,
    [_i, _s, _sT, _eT, _co, _de],
    [0, 0, 5, 5, () => DestinationConfiguration, () => DestinationDetail],
];
var DestinationConfiguration = [
    3,
    n0,
    _DC,
    0,
    [_n, _ch, _s_],
    [0, () => ChannelDestinationConfiguration, () => S3DestinationConfiguration],
];
var DestinationDetail = [3, n0, _DD, 0, [_s_], [() => S3Detail]];
var DestinationSummary = [3, n0, _DS, 0, [_i, _s, _sT, _eT], [0, 0, 5, 5]];
var DisconnectParticipantRequest = [3, n0, _DPR, 0, [_sA, _pI, _r], [0, 0, 0]];
var DisconnectParticipantResponse = [3, n0, _DPRi, 0, [], []];
var EncoderConfiguration = [
    3,
    n0,
    _EC,
    0,
    [_a, _n, _v, _t],
    [0, 0, () => Video, 128 | 0],
];
var EncoderConfigurationSummary = [3, n0, _ECS, 0, [_a, _n, _t], [0, 0, 128 | 0]];
var Event = [
    3,
    n0,
    _E,
    0,
    [_n, _pI, _eTv, _rPI, _eCr, _dSA, _dSI, _re],
    [0, 0, 5, 0, 0, 0, 0, 2],
];
var GetCompositionRequest = [3, n0, _GCR, 0, [_a], [0]];
var GetCompositionResponse = [3, n0, _GCRe, 0, [_com], [() => Composition]];
var GetEncoderConfigurationRequest = [3, n0, _GECR, 0, [_a], [0]];
var GetEncoderConfigurationResponse = [
    3,
    n0,
    _GECRe,
    0,
    [_eC],
    [() => EncoderConfiguration],
];
var GetIngestConfigurationRequest = [3, n0, _GICR, 0, [_a], [0]];
var GetIngestConfigurationResponse = [
    3,
    n0,
    _GICRe,
    0,
    [_iC],
    [[() => IngestConfiguration, 0]],
];
var GetParticipantRequest = [3, n0, _GPR, 0, [_sA, _sI, _pI], [0, 0, 0]];
var GetParticipantResponse = [3, n0, _GPRe, 0, [_p], [() => Participant]];
var GetPublicKeyRequest = [3, n0, _GPKR, 0, [_a], [0]];
var GetPublicKeyResponse = [3, n0, _GPKRe, 0, [_pK], [() => PublicKey]];
var GetStageRequest = [3, n0, _GSR, 0, [_a], [0]];
var GetStageResponse = [3, n0, _GSRe, 0, [_sta], [() => Stage]];
var GetStageSessionRequest = [3, n0, _GSSR, 0, [_sA, _sI], [0, 0]];
var GetStageSessionResponse = [3, n0, _GSSRe, 0, [_sS], [() => StageSession]];
var GetStorageConfigurationRequest = [3, n0, _GSCR, 0, [_a], [0]];
var GetStorageConfigurationResponse = [
    3,
    n0,
    _GSCRe,
    0,
    [_sC],
    [() => StorageConfiguration],
];
var GridConfiguration = [
    3,
    n0,
    _GC,
    0,
    [_fPA, _oSV, _vAR, _vFM, _gG, _pOA],
    [0, 2, 0, 0, 1, 0],
];
var ImportPublicKeyRequest = [3, n0, _IPKR, 0, [_pKM, _n, _t], [0, 0, 128 | 0]];
var ImportPublicKeyResponse = [3, n0, _IPKRm, 0, [_pK], [() => PublicKey]];
var IngestConfiguration = [
    3,
    n0,
    _IC,
    0,
    [_n, _a, _iP, _sK, _sA, _pI, _s, _uI, _at, _t],
    [0, 0, 0, [() => StreamKey, 0], 0, 0, 0, 0, 128 | 0, 128 | 0],
];
var IngestConfigurationSummary = [
    3,
    n0,
    _ICS,
    0,
    [_n, _a, _iP, _sA, _pI, _s, _uI],
    [0, 0, 0, 0, 0, 0, 0],
];
var InternalServerException = [
    -3,
    n0,
    _ISE,
    {
        [_e]: _se,
        [_hE]: 500,
    },
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO, _xAET, _eM],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
        [
            0,
            {
                [_hH]: _xaE,
            },
        ],
        0,
    ],
];
schema.TypeRegistry.for(n0).registerError(InternalServerException, InternalServerException$1);
var LayoutConfiguration = [
    3,
    n0,
    _LC,
    0,
    [_g, _pi],
    [() => GridConfiguration, () => PipConfiguration],
];
var ListCompositionsRequest = [3, n0, _LCR, 0, [_fBSA, _fBECA, _nT, _mR], [0, 0, 0, 1]];
var ListCompositionsResponse = [
    3,
    n0,
    _LCRi,
    0,
    [_comp, _nT],
    [() => CompositionSummaryList, 0],
];
var ListEncoderConfigurationsRequest = [3, n0, _LECR, 0, [_nT, _mR], [0, 1]];
var ListEncoderConfigurationsResponse = [
    3,
    n0,
    _LECRi,
    0,
    [_eCn, _nT],
    [() => EncoderConfigurationSummaryList, 0],
];
var ListIngestConfigurationsRequest = [
    3,
    n0,
    _LICR,
    0,
    [_fBSA, _fBS, _nT, _mR],
    [0, 0, 0, 1],
];
var ListIngestConfigurationsResponse = [
    3,
    n0,
    _LICRi,
    0,
    [_iCn, _nT],
    [() => IngestConfigurationList, 0],
];
var ListParticipantEventsRequest = [
    3,
    n0,
    _LPER,
    0,
    [_sA, _sI, _pI, _nT, _mR],
    [0, 0, 0, 0, 1],
];
var ListParticipantEventsResponse = [3, n0, _LPERi, 0, [_ev, _nT], [() => EventList, 0]];
var ListParticipantReplicasRequest = [
    3,
    n0,
    _LPRR,
    0,
    [_sSA, _pI, _nT, _mR],
    [0, 0, 0, 1],
];
var ListParticipantReplicasResponse = [
    3,
    n0,
    _LPRRi,
    0,
    [_rep, _nT],
    [() => ParticipantReplicaList, 0],
];
var ListParticipantsRequest = [
    3,
    n0,
    _LPR,
    0,
    [_sA, _sI, _fBUI, _fBP, _fBS, _nT, _mR, _fBRS],
    [0, 0, 0, 2, 0, 0, 1, 0],
];
var ListParticipantsResponse = [3, n0, _LPRi, 0, [_pa, _nT], [() => ParticipantList, 0]];
var ListPublicKeysRequest = [3, n0, _LPKR, 0, [_nT, _mR], [0, 1]];
var ListPublicKeysResponse = [3, n0, _LPKRi, 0, [_pKu, _nT], [() => PublicKeyList, 0]];
var ListStageSessionsRequest = [3, n0, _LSSR, 0, [_sA, _nT, _mR], [0, 0, 1]];
var ListStageSessionsResponse = [
    3,
    n0,
    _LSSRi,
    0,
    [_sSt, _nT],
    [() => StageSessionList, 0],
];
var ListStagesRequest = [3, n0, _LSR, 0, [_nT, _mR], [0, 1]];
var ListStagesResponse = [3, n0, _LSRi, 0, [_stag, _nT], [() => StageSummaryList, 0]];
var ListStorageConfigurationsRequest = [3, n0, _LSCR, 0, [_nT, _mR], [0, 1]];
var ListStorageConfigurationsResponse = [
    3,
    n0,
    _LSCRi,
    0,
    [_sCt, _nT],
    [() => StorageConfigurationSummaryList, 0],
];
var ListTagsForResourceRequest = [3, n0, _LTFRR, 0, [_rA], [[0, 1]]];
var ListTagsForResourceResponse = [3, n0, _LTFRRi, 0, [_t], [128 | 0]];
var Participant = [
    3,
    n0,
    _P,
    0,
    [_pI, _uI, _s, _fJT, _at, _pu, _iN, _oN, _oV, _bN, _bV, _sV, _rSBN, _rSP, _rS, _pr, _rT, _rSe, _sSA, _sSI],
    [0, 0, 0, 5, 128 | 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
];
var ParticipantRecordingHlsConfiguration = [3, n0, _PRHC, 0, [_tSDS], [1]];
var ParticipantReplica = [
    3,
    n0,
    _PR,
    0,
    [_sSA, _pI, _sSI, _dSA, _dSI, _rSe],
    [0, 0, 0, 0, 0, 0],
];
var ParticipantSummary = [
    3,
    n0,
    _PS,
    0,
    [_pI, _uI, _s, _fJT, _pu, _rS, _rT, _rSe, _sSA, _sSI],
    [0, 0, 0, 5, 2, 0, 0, 0, 0, 0],
];
var ParticipantThumbnailConfiguration = [
    3,
    n0,
    _PTC,
    0,
    [_tIS, _st, _rM],
    [1, 64 | 0, 0],
];
var ParticipantToken = [
    3,
    n0,
    _PT,
    0,
    [_pI, _to, _uI, _at, _du, _ca, _eTx],
    [0, [() => ParticipantTokenString, 0], 0, 128 | 0, 1, 64 | 0, 5],
];
var ParticipantTokenConfiguration = [
    3,
    n0,
    _PTCa,
    0,
    [_du, _uI, _at, _ca],
    [1, 0, 128 | 0, 64 | 0],
];
var PendingVerification = [
    -3,
    n0,
    _PV,
    {
        [_e]: _c,
        [_hE]: 403,
    },
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO, _xAET, _eM],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
        [
            0,
            {
                [_hH]: _xaE,
            },
        ],
        0,
    ],
];
schema.TypeRegistry.for(n0).registerError(PendingVerification, PendingVerification$1);
var PipConfiguration = [
    3,
    n0,
    _PC,
    0,
    [_fPA, _oSV, _vFM, _gG, _pPA, _pB, _pO, _pP, _pW, _pH, _pOA],
    [0, 2, 0, 1, 0, 0, 1, 0, 1, 1, 0],
];
var PublicKey = [3, n0, _PK, 0, [_a, _n, _pKM, _fi, _t], [0, 0, 0, 0, 128 | 0]];
var PublicKeySummary = [3, n0, _PKS, 0, [_a, _n, _t], [0, 0, 128 | 0]];
var RecordingConfiguration = [
    3,
    n0,
    _RC,
    0,
    [_hC, _fo],
    [() => CompositionRecordingHlsConfiguration, 0],
];
var ResourceNotFoundException = [
    -3,
    n0,
    _RNFE,
    {
        [_e]: _c,
        [_hE]: 404,
    },
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO, _xAET, _eM],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
        [
            0,
            {
                [_hH]: _xaE,
            },
        ],
        0,
    ],
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException, ResourceNotFoundException$1);
var S3DestinationConfiguration = [
    3,
    n0,
    _SDC,
    0,
    [_sCA, _eCAn, _rC, _tCh],
    [0, 64 | 0, () => RecordingConfiguration, () => CompositionThumbnailConfigurationList],
];
var S3Detail = [3, n0, _SD, 0, [_rP], [0]];
var S3StorageConfiguration = [3, n0, _SSC, 0, [_bNu], [0]];
var ServiceQuotaExceededException = [
    -3,
    n0,
    _SQEE,
    {
        [_e]: _c,
        [_hE]: 402,
    },
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO, _xAET, _eM],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
        [
            0,
            {
                [_hH]: _xaE,
            },
        ],
        0,
    ],
];
schema.TypeRegistry.for(n0).registerError(ServiceQuotaExceededException, ServiceQuotaExceededException$1);
var Stage = [
    3,
    n0,
    _S,
    0,
    [_a, _n, _aSI, _t, _aPRC, _en],
    [0, 0, 0, 128 | 0, () => AutoParticipantRecordingConfiguration, () => StageEndpoints],
];
var StageEndpoints = [3, n0, _SE, 0, [_ev, _w, _rt, _rtm], [0, 0, 0, 0]];
var StageSession = [3, n0, _SS, 0, [_sI, _sT, _eT], [0, 5, 5]];
var StageSessionSummary = [3, n0, _SSS, 0, [_sI, _sT, _eT], [0, 5, 5]];
var StageSummary = [3, n0, _SSt, 0, [_a, _n, _aSI, _t], [0, 0, 0, 128 | 0]];
var StartCompositionRequest = [
    3,
    n0,
    _SCR,
    0,
    [_sA, _iT, _l, _d, _t],
    [0, [0, 4], () => LayoutConfiguration, () => DestinationConfigurationList, 128 | 0],
];
var StartCompositionResponse = [3, n0, _SCRt, 0, [_com], [() => Composition]];
var StartParticipantReplicationRequest = [
    3,
    n0,
    _SPRR,
    0,
    [_sSA, _dSA, _pI, _rWS, _at],
    [0, 0, 0, 1, 128 | 0],
];
var StartParticipantReplicationResponse = [
    3,
    n0,
    _SPRRt,
    0,
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
    ],
];
var StopCompositionRequest = [3, n0, _SCRto, 0, [_a], [0]];
var StopCompositionResponse = [3, n0, _SCRtop, 0, [], []];
var StopParticipantReplicationRequest = [3, n0, _SPRRto, 0, [_sSA, _dSA, _pI], [0, 0, 0]];
var StopParticipantReplicationResponse = [
    3,
    n0,
    _SPRRtop,
    0,
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
    ],
];
var StorageConfiguration = [
    3,
    n0,
    _SC,
    0,
    [_a, _n, _s_, _t],
    [0, 0, () => S3StorageConfiguration, 128 | 0],
];
var StorageConfigurationSummary = [
    3,
    n0,
    _SCS,
    0,
    [_a, _n, _s_, _t],
    [0, 0, () => S3StorageConfiguration, 128 | 0],
];
var TagResourceRequest = [3, n0, _TRR, 0, [_rA, _t], [[0, 1], 128 | 0]];
var TagResourceResponse = [3, n0, _TRRa, 0, [], []];
var UntagResourceRequest = [
    3,
    n0,
    _URR,
    0,
    [_rA, _tK],
    [
        [0, 1],
        [
            64 | 0,
            {
                [_hQ]: _tK,
            },
        ],
    ],
];
var UntagResourceResponse = [3, n0, _URRn, 0, [], []];
var UpdateIngestConfigurationRequest = [3, n0, _UICR, 0, [_a, _sA], [0, 0]];
var UpdateIngestConfigurationResponse = [
    3,
    n0,
    _UICRp,
    0,
    [_iC],
    [[() => IngestConfiguration, 0]],
];
var UpdateStageRequest = [
    3,
    n0,
    _USR,
    0,
    [_a, _n, _aPRC],
    [0, 0, () => AutoParticipantRecordingConfiguration],
];
var UpdateStageResponse = [3, n0, _USRp, 0, [_sta], [() => Stage]];
var ValidationException = [
    -3,
    n0,
    _VE,
    {
        [_e]: _c,
        [_hE]: 400,
    },
    [_aCAO, _aCEH, _cC, _cSP, _sTS, _xCTO, _xFO, _xAET, _eM],
    [
        [
            0,
            {
                [_hH]: _ACAO,
            },
        ],
        [
            0,
            {
                [_hH]: _ACEH,
            },
        ],
        [
            0,
            {
                [_hH]: _CC,
            },
        ],
        [
            0,
            {
                [_hH]: _CSP,
            },
        ],
        [
            0,
            {
                [_hH]: _STS,
            },
        ],
        [
            0,
            {
                [_hH]: _XCTO,
            },
        ],
        [
            0,
            {
                [_hH]: _XFO,
            },
        ],
        [
            0,
            {
                [_hH]: _xaE,
            },
        ],
        0,
    ],
];
schema.TypeRegistry.for(n0).registerError(ValidationException, ValidationException$1);
var Video = [3, n0, _V, 0, [_wi, _h, _fr, _b], [1, 1, 1, 1]];
var IVSRealTimeServiceException = [-3, _sm, "IVSRealTimeServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(IVSRealTimeServiceException, IVSRealTimeServiceException$1);
var CompositionSummaryList = [1, n0, _CSL, 0, () => CompositionSummary];
var CompositionThumbnailConfigurationList = [
    1,
    n0,
    _CTCL,
    0,
    () => CompositionThumbnailConfiguration,
];
var DestinationConfigurationList = [1, n0, _DCL, 0, () => DestinationConfiguration];
var DestinationList = [1, n0, _DL, 0, () => Destination];
var DestinationSummaryList = [1, n0, _DSL, 0, () => DestinationSummary];
var EncoderConfigurationSummaryList = [1, n0, _ECSL, 0, () => EncoderConfigurationSummary];
var EventList = [1, n0, _EL, 0, () => Event];
var IngestConfigurationList = [1, n0, _ICL, 0, () => IngestConfigurationSummary];
var ParticipantList = [1, n0, _PL, 0, () => ParticipantSummary];
var ParticipantReplicaList = [1, n0, _PRL, 0, () => ParticipantReplica];
var ParticipantTokenConfigurations = [1, n0, _PTCar, 0, () => ParticipantTokenConfiguration];
var ParticipantTokenList = [1, n0, _PTL, 0, [() => ParticipantToken, 0]];
var PublicKeyList = [1, n0, _PKL, 0, () => PublicKeySummary];
var StageSessionList = [1, n0, _SSL, 0, () => StageSessionSummary];
var StageSummaryList = [1, n0, _SSLt, 0, () => StageSummary];
var StorageConfigurationSummaryList = [1, n0, _SCSL, 0, () => StorageConfigurationSummary];
var CreateEncoderConfiguration = [
    9,
    n0,
    _CEC,
    {
        [_ht]: ["POST", "/CreateEncoderConfiguration", 200],
    },
    () => CreateEncoderConfigurationRequest,
    () => CreateEncoderConfigurationResponse,
];
var CreateIngestConfiguration = [
    9,
    n0,
    _CIC,
    {
        [_ht]: ["POST", "/CreateIngestConfiguration", 200],
    },
    () => CreateIngestConfigurationRequest,
    () => CreateIngestConfigurationResponse,
];
var CreateParticipantToken = [
    9,
    n0,
    _CPT,
    {
        [_ht]: ["POST", "/CreateParticipantToken", 200],
    },
    () => CreateParticipantTokenRequest,
    () => CreateParticipantTokenResponse,
];
var CreateStage = [
    9,
    n0,
    _CSr,
    {
        [_ht]: ["POST", "/CreateStage", 200],
    },
    () => CreateStageRequest,
    () => CreateStageResponse,
];
var CreateStorageConfiguration = [
    9,
    n0,
    _CSC,
    {
        [_ht]: ["POST", "/CreateStorageConfiguration", 200],
    },
    () => CreateStorageConfigurationRequest,
    () => CreateStorageConfigurationResponse,
];
var DeleteEncoderConfiguration = [
    9,
    n0,
    _DEC,
    {
        [_ht]: ["POST", "/DeleteEncoderConfiguration", 200],
    },
    () => DeleteEncoderConfigurationRequest,
    () => DeleteEncoderConfigurationResponse,
];
var DeleteIngestConfiguration = [
    9,
    n0,
    _DIC,
    {
        [_ht]: ["POST", "/DeleteIngestConfiguration", 200],
    },
    () => DeleteIngestConfigurationRequest,
    () => DeleteIngestConfigurationResponse,
];
var DeletePublicKey = [
    9,
    n0,
    _DPK,
    {
        [_ht]: ["POST", "/DeletePublicKey", 200],
    },
    () => DeletePublicKeyRequest,
    () => DeletePublicKeyResponse,
];
var DeleteStage = [
    9,
    n0,
    _DSe,
    {
        [_ht]: ["POST", "/DeleteStage", 200],
    },
    () => DeleteStageRequest,
    () => DeleteStageResponse,
];
var DeleteStorageConfiguration = [
    9,
    n0,
    _DSC,
    {
        [_ht]: ["POST", "/DeleteStorageConfiguration", 200],
    },
    () => DeleteStorageConfigurationRequest,
    () => DeleteStorageConfigurationResponse,
];
var DisconnectParticipant = [
    9,
    n0,
    _DP,
    {
        [_ht]: ["POST", "/DisconnectParticipant", 200],
    },
    () => DisconnectParticipantRequest,
    () => DisconnectParticipantResponse,
];
var GetComposition = [
    9,
    n0,
    _GCe,
    {
        [_ht]: ["POST", "/GetComposition", 200],
    },
    () => GetCompositionRequest,
    () => GetCompositionResponse,
];
var GetEncoderConfiguration = [
    9,
    n0,
    _GEC,
    {
        [_ht]: ["POST", "/GetEncoderConfiguration", 200],
    },
    () => GetEncoderConfigurationRequest,
    () => GetEncoderConfigurationResponse,
];
var GetIngestConfiguration = [
    9,
    n0,
    _GIC,
    {
        [_ht]: ["POST", "/GetIngestConfiguration", 200],
    },
    () => GetIngestConfigurationRequest,
    () => GetIngestConfigurationResponse,
];
var GetParticipant = [
    9,
    n0,
    _GP,
    {
        [_ht]: ["POST", "/GetParticipant", 200],
    },
    () => GetParticipantRequest,
    () => GetParticipantResponse,
];
var GetPublicKey = [
    9,
    n0,
    _GPK,
    {
        [_ht]: ["POST", "/GetPublicKey", 200],
    },
    () => GetPublicKeyRequest,
    () => GetPublicKeyResponse,
];
var GetStage = [
    9,
    n0,
    _GS,
    {
        [_ht]: ["POST", "/GetStage", 200],
    },
    () => GetStageRequest,
    () => GetStageResponse,
];
var GetStageSession = [
    9,
    n0,
    _GSS,
    {
        [_ht]: ["POST", "/GetStageSession", 200],
    },
    () => GetStageSessionRequest,
    () => GetStageSessionResponse,
];
var GetStorageConfiguration = [
    9,
    n0,
    _GSC,
    {
        [_ht]: ["POST", "/GetStorageConfiguration", 200],
    },
    () => GetStorageConfigurationRequest,
    () => GetStorageConfigurationResponse,
];
var ImportPublicKey = [
    9,
    n0,
    _IPK,
    {
        [_ht]: ["POST", "/ImportPublicKey", 200],
    },
    () => ImportPublicKeyRequest,
    () => ImportPublicKeyResponse,
];
var ListCompositions = [
    9,
    n0,
    _LCi,
    {
        [_ht]: ["POST", "/ListCompositions", 200],
    },
    () => ListCompositionsRequest,
    () => ListCompositionsResponse,
];
var ListEncoderConfigurations = [
    9,
    n0,
    _LEC,
    {
        [_ht]: ["POST", "/ListEncoderConfigurations", 200],
    },
    () => ListEncoderConfigurationsRequest,
    () => ListEncoderConfigurationsResponse,
];
var ListIngestConfigurations = [
    9,
    n0,
    _LIC,
    {
        [_ht]: ["POST", "/ListIngestConfigurations", 200],
    },
    () => ListIngestConfigurationsRequest,
    () => ListIngestConfigurationsResponse,
];
var ListParticipantEvents = [
    9,
    n0,
    _LPE,
    {
        [_ht]: ["POST", "/ListParticipantEvents", 200],
    },
    () => ListParticipantEventsRequest,
    () => ListParticipantEventsResponse,
];
var ListParticipantReplicas = [
    9,
    n0,
    _LPRis,
    {
        [_ht]: ["POST", "/ListParticipantReplicas", 200],
    },
    () => ListParticipantReplicasRequest,
    () => ListParticipantReplicasResponse,
];
var ListParticipants = [
    9,
    n0,
    _LP,
    {
        [_ht]: ["POST", "/ListParticipants", 200],
    },
    () => ListParticipantsRequest,
    () => ListParticipantsResponse,
];
var ListPublicKeys = [
    9,
    n0,
    _LPK,
    {
        [_ht]: ["POST", "/ListPublicKeys", 200],
    },
    () => ListPublicKeysRequest,
    () => ListPublicKeysResponse,
];
var ListStages = [
    9,
    n0,
    _LS,
    {
        [_ht]: ["POST", "/ListStages", 200],
    },
    () => ListStagesRequest,
    () => ListStagesResponse,
];
var ListStageSessions = [
    9,
    n0,
    _LSS,
    {
        [_ht]: ["POST", "/ListStageSessions", 200],
    },
    () => ListStageSessionsRequest,
    () => ListStageSessionsResponse,
];
var ListStorageConfigurations = [
    9,
    n0,
    _LSC,
    {
        [_ht]: ["POST", "/ListStorageConfigurations", 200],
    },
    () => ListStorageConfigurationsRequest,
    () => ListStorageConfigurationsResponse,
];
var ListTagsForResource = [
    9,
    n0,
    _LTFR,
    {
        [_ht]: ["GET", "/tags/{resourceArn}", 200],
    },
    () => ListTagsForResourceRequest,
    () => ListTagsForResourceResponse,
];
var StartComposition = [
    9,
    n0,
    _SCt,
    {
        [_ht]: ["POST", "/StartComposition", 200],
    },
    () => StartCompositionRequest,
    () => StartCompositionResponse,
];
var StartParticipantReplication = [
    9,
    n0,
    _SPR,
    {
        [_ht]: ["POST", "/StartParticipantReplication", 200],
    },
    () => StartParticipantReplicationRequest,
    () => StartParticipantReplicationResponse,
];
var StopComposition = [
    9,
    n0,
    _SCto,
    {
        [_ht]: ["POST", "/StopComposition", 200],
    },
    () => StopCompositionRequest,
    () => StopCompositionResponse,
];
var StopParticipantReplication = [
    9,
    n0,
    _SPRt,
    {
        [_ht]: ["POST", "/StopParticipantReplication", 200],
    },
    () => StopParticipantReplicationRequest,
    () => StopParticipantReplicationResponse,
];
var TagResource = [
    9,
    n0,
    _TR,
    {
        [_ht]: ["POST", "/tags/{resourceArn}", 200],
    },
    () => TagResourceRequest,
    () => TagResourceResponse,
];
var UntagResource = [
    9,
    n0,
    _UR,
    {
        [_ht]: ["DELETE", "/tags/{resourceArn}", 200],
    },
    () => UntagResourceRequest,
    () => UntagResourceResponse,
];
var UpdateIngestConfiguration = [
    9,
    n0,
    _UIC,
    {
        [_ht]: ["POST", "/UpdateIngestConfiguration", 200],
    },
    () => UpdateIngestConfigurationRequest,
    () => UpdateIngestConfigurationResponse,
];
var UpdateStage = [
    9,
    n0,
    _US,
    {
        [_ht]: ["POST", "/UpdateStage", 200],
    },
    () => UpdateStageRequest,
    () => UpdateStageResponse,
];

class CreateEncoderConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "CreateEncoderConfiguration", {})
    .n("IVSRealTimeClient", "CreateEncoderConfigurationCommand")
    .sc(CreateEncoderConfiguration)
    .build() {
}

class CreateIngestConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "CreateIngestConfiguration", {})
    .n("IVSRealTimeClient", "CreateIngestConfigurationCommand")
    .sc(CreateIngestConfiguration)
    .build() {
}

class CreateParticipantTokenCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "CreateParticipantToken", {})
    .n("IVSRealTimeClient", "CreateParticipantTokenCommand")
    .sc(CreateParticipantToken)
    .build() {
}

class CreateStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "CreateStage", {})
    .n("IVSRealTimeClient", "CreateStageCommand")
    .sc(CreateStage)
    .build() {
}

class CreateStorageConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "CreateStorageConfiguration", {})
    .n("IVSRealTimeClient", "CreateStorageConfigurationCommand")
    .sc(CreateStorageConfiguration)
    .build() {
}

class DeleteEncoderConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "DeleteEncoderConfiguration", {})
    .n("IVSRealTimeClient", "DeleteEncoderConfigurationCommand")
    .sc(DeleteEncoderConfiguration)
    .build() {
}

class DeleteIngestConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "DeleteIngestConfiguration", {})
    .n("IVSRealTimeClient", "DeleteIngestConfigurationCommand")
    .sc(DeleteIngestConfiguration)
    .build() {
}

class DeletePublicKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "DeletePublicKey", {})
    .n("IVSRealTimeClient", "DeletePublicKeyCommand")
    .sc(DeletePublicKey)
    .build() {
}

class DeleteStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "DeleteStage", {})
    .n("IVSRealTimeClient", "DeleteStageCommand")
    .sc(DeleteStage)
    .build() {
}

class DeleteStorageConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "DeleteStorageConfiguration", {})
    .n("IVSRealTimeClient", "DeleteStorageConfigurationCommand")
    .sc(DeleteStorageConfiguration)
    .build() {
}

class DisconnectParticipantCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "DisconnectParticipant", {})
    .n("IVSRealTimeClient", "DisconnectParticipantCommand")
    .sc(DisconnectParticipant)
    .build() {
}

class GetCompositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetComposition", {})
    .n("IVSRealTimeClient", "GetCompositionCommand")
    .sc(GetComposition)
    .build() {
}

class GetEncoderConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetEncoderConfiguration", {})
    .n("IVSRealTimeClient", "GetEncoderConfigurationCommand")
    .sc(GetEncoderConfiguration)
    .build() {
}

class GetIngestConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetIngestConfiguration", {})
    .n("IVSRealTimeClient", "GetIngestConfigurationCommand")
    .sc(GetIngestConfiguration)
    .build() {
}

class GetParticipantCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetParticipant", {})
    .n("IVSRealTimeClient", "GetParticipantCommand")
    .sc(GetParticipant)
    .build() {
}

class GetPublicKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetPublicKey", {})
    .n("IVSRealTimeClient", "GetPublicKeyCommand")
    .sc(GetPublicKey)
    .build() {
}

class GetStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetStage", {})
    .n("IVSRealTimeClient", "GetStageCommand")
    .sc(GetStage)
    .build() {
}

class GetStageSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetStageSession", {})
    .n("IVSRealTimeClient", "GetStageSessionCommand")
    .sc(GetStageSession)
    .build() {
}

class GetStorageConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "GetStorageConfiguration", {})
    .n("IVSRealTimeClient", "GetStorageConfigurationCommand")
    .sc(GetStorageConfiguration)
    .build() {
}

class ImportPublicKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ImportPublicKey", {})
    .n("IVSRealTimeClient", "ImportPublicKeyCommand")
    .sc(ImportPublicKey)
    .build() {
}

class ListCompositionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListCompositions", {})
    .n("IVSRealTimeClient", "ListCompositionsCommand")
    .sc(ListCompositions)
    .build() {
}

class ListEncoderConfigurationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListEncoderConfigurations", {})
    .n("IVSRealTimeClient", "ListEncoderConfigurationsCommand")
    .sc(ListEncoderConfigurations)
    .build() {
}

class ListIngestConfigurationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListIngestConfigurations", {})
    .n("IVSRealTimeClient", "ListIngestConfigurationsCommand")
    .sc(ListIngestConfigurations)
    .build() {
}

class ListParticipantEventsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListParticipantEvents", {})
    .n("IVSRealTimeClient", "ListParticipantEventsCommand")
    .sc(ListParticipantEvents)
    .build() {
}

class ListParticipantReplicasCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListParticipantReplicas", {})
    .n("IVSRealTimeClient", "ListParticipantReplicasCommand")
    .sc(ListParticipantReplicas)
    .build() {
}

class ListParticipantsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListParticipants", {})
    .n("IVSRealTimeClient", "ListParticipantsCommand")
    .sc(ListParticipants)
    .build() {
}

class ListPublicKeysCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListPublicKeys", {})
    .n("IVSRealTimeClient", "ListPublicKeysCommand")
    .sc(ListPublicKeys)
    .build() {
}

class ListStagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListStages", {})
    .n("IVSRealTimeClient", "ListStagesCommand")
    .sc(ListStages)
    .build() {
}

class ListStageSessionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListStageSessions", {})
    .n("IVSRealTimeClient", "ListStageSessionsCommand")
    .sc(ListStageSessions)
    .build() {
}

class ListStorageConfigurationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListStorageConfigurations", {})
    .n("IVSRealTimeClient", "ListStorageConfigurationsCommand")
    .sc(ListStorageConfigurations)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "ListTagsForResource", {})
    .n("IVSRealTimeClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource)
    .build() {
}

class StartCompositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "StartComposition", {})
    .n("IVSRealTimeClient", "StartCompositionCommand")
    .sc(StartComposition)
    .build() {
}

class StartParticipantReplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "StartParticipantReplication", {})
    .n("IVSRealTimeClient", "StartParticipantReplicationCommand")
    .sc(StartParticipantReplication)
    .build() {
}

class StopCompositionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "StopComposition", {})
    .n("IVSRealTimeClient", "StopCompositionCommand")
    .sc(StopComposition)
    .build() {
}

class StopParticipantReplicationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "StopParticipantReplication", {})
    .n("IVSRealTimeClient", "StopParticipantReplicationCommand")
    .sc(StopParticipantReplication)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "TagResource", {})
    .n("IVSRealTimeClient", "TagResourceCommand")
    .sc(TagResource)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "UntagResource", {})
    .n("IVSRealTimeClient", "UntagResourceCommand")
    .sc(UntagResource)
    .build() {
}

class UpdateIngestConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "UpdateIngestConfiguration", {})
    .n("IVSRealTimeClient", "UpdateIngestConfigurationCommand")
    .sc(UpdateIngestConfiguration)
    .build() {
}

class UpdateStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonInteractiveVideoServiceRealTime", "UpdateStage", {})
    .n("IVSRealTimeClient", "UpdateStageCommand")
    .sc(UpdateStage)
    .build() {
}

const commands = {
    CreateEncoderConfigurationCommand,
    CreateIngestConfigurationCommand,
    CreateParticipantTokenCommand,
    CreateStageCommand,
    CreateStorageConfigurationCommand,
    DeleteEncoderConfigurationCommand,
    DeleteIngestConfigurationCommand,
    DeletePublicKeyCommand,
    DeleteStageCommand,
    DeleteStorageConfigurationCommand,
    DisconnectParticipantCommand,
    GetCompositionCommand,
    GetEncoderConfigurationCommand,
    GetIngestConfigurationCommand,
    GetParticipantCommand,
    GetPublicKeyCommand,
    GetStageCommand,
    GetStageSessionCommand,
    GetStorageConfigurationCommand,
    ImportPublicKeyCommand,
    ListCompositionsCommand,
    ListEncoderConfigurationsCommand,
    ListIngestConfigurationsCommand,
    ListParticipantEventsCommand,
    ListParticipantReplicasCommand,
    ListParticipantsCommand,
    ListPublicKeysCommand,
    ListStagesCommand,
    ListStageSessionsCommand,
    ListStorageConfigurationsCommand,
    ListTagsForResourceCommand,
    StartCompositionCommand,
    StartParticipantReplicationCommand,
    StopCompositionCommand,
    StopParticipantReplicationCommand,
    TagResourceCommand,
    UntagResourceCommand,
    UpdateIngestConfigurationCommand,
    UpdateStageCommand,
};
class IVSRealTime extends IVSRealTimeClient {
}
smithyClient.createAggregatedClient(commands, IVSRealTime);

const paginateListCompositions = core.createPaginator(IVSRealTimeClient, ListCompositionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListEncoderConfigurations = core.createPaginator(IVSRealTimeClient, ListEncoderConfigurationsCommand, "nextToken", "nextToken", "maxResults");

const paginateListIngestConfigurations = core.createPaginator(IVSRealTimeClient, ListIngestConfigurationsCommand, "nextToken", "nextToken", "maxResults");

const paginateListParticipantEvents = core.createPaginator(IVSRealTimeClient, ListParticipantEventsCommand, "nextToken", "nextToken", "maxResults");

const paginateListParticipantReplicas = core.createPaginator(IVSRealTimeClient, ListParticipantReplicasCommand, "nextToken", "nextToken", "maxResults");

const paginateListParticipants = core.createPaginator(IVSRealTimeClient, ListParticipantsCommand, "nextToken", "nextToken", "maxResults");

const paginateListPublicKeys = core.createPaginator(IVSRealTimeClient, ListPublicKeysCommand, "nextToken", "nextToken", "maxResults");

const paginateListStageSessions = core.createPaginator(IVSRealTimeClient, ListStageSessionsCommand, "nextToken", "nextToken", "maxResults");

const paginateListStages = core.createPaginator(IVSRealTimeClient, ListStagesCommand, "nextToken", "nextToken", "maxResults");

const paginateListStorageConfigurations = core.createPaginator(IVSRealTimeClient, ListStorageConfigurationsCommand, "nextToken", "nextToken", "maxResults");

const IngestProtocol = {
    RTMP: "RTMP",
    RTMPS: "RTMPS",
};
const ParticipantRecordingMediaType = {
    AUDIO_ONLY: "AUDIO_ONLY",
    AUDIO_VIDEO: "AUDIO_VIDEO",
    NONE: "NONE",
};
const ThumbnailRecordingMode = {
    DISABLED: "DISABLED",
    INTERVAL: "INTERVAL",
};
const ThumbnailStorageType = {
    LATEST: "LATEST",
    SEQUENTIAL: "SEQUENTIAL",
};
const VideoAspectRatio = {
    AUTO: "AUTO",
    PORTRAIT: "PORTRAIT",
    SQUARE: "SQUARE",
    VIDEO: "VIDEO",
};
const VideoFillMode = {
    CONTAIN: "CONTAIN",
    COVER: "COVER",
    FILL: "FILL",
};
const PipBehavior = {
    DYNAMIC: "DYNAMIC",
    STATIC: "STATIC",
};
const PipPosition = {
    BOTTOM_LEFT: "BOTTOM_LEFT",
    BOTTOM_RIGHT: "BOTTOM_RIGHT",
    TOP_LEFT: "TOP_LEFT",
    TOP_RIGHT: "TOP_RIGHT",
};
const ParticipantProtocol = {
    RTMP: "RTMP",
    RTMPS: "RTMPS",
    UNKNOWN: "UNKNOWN",
    WHIP: "WHIP",
};
const EventErrorCode = {
    BITRATE_EXCEEDED: "BITRATE_EXCEEDED",
    B_FRAME_PRESENT: "B_FRAME_PRESENT",
    INSUFFICIENT_CAPABILITIES: "INSUFFICIENT_CAPABILITIES",
    INTERNAL_SERVER_EXCEPTION: "INTERNAL_SERVER_EXCEPTION",
    INVALID_AUDIO_CODEC: "INVALID_AUDIO_CODEC",
    INVALID_INPUT: "INVALID_INPUT",
    INVALID_PROTOCOL: "INVALID_PROTOCOL",
    INVALID_STREAM_KEY: "INVALID_STREAM_KEY",
    INVALID_VIDEO_CODEC: "INVALID_VIDEO_CODEC",
    PUBLISHER_NOT_FOUND: "PUBLISHER_NOT_FOUND",
    QUOTA_EXCEEDED: "QUOTA_EXCEEDED",
    RESOLUTION_EXCEEDED: "RESOLUTION_EXCEEDED",
    REUSE_OF_STREAM_KEY: "REUSE_OF_STREAM_KEY",
    STREAM_DURATION_EXCEEDED: "STREAM_DURATION_EXCEEDED",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.AccessDeniedException = AccessDeniedException$1;
exports.ConflictException = ConflictException$1;
exports.CreateEncoderConfigurationCommand = CreateEncoderConfigurationCommand;
exports.CreateIngestConfigurationCommand = CreateIngestConfigurationCommand;
exports.CreateParticipantTokenCommand = CreateParticipantTokenCommand;
exports.CreateStageCommand = CreateStageCommand;
exports.CreateStorageConfigurationCommand = CreateStorageConfigurationCommand;
exports.DeleteEncoderConfigurationCommand = DeleteEncoderConfigurationCommand;
exports.DeleteIngestConfigurationCommand = DeleteIngestConfigurationCommand;
exports.DeletePublicKeyCommand = DeletePublicKeyCommand;
exports.DeleteStageCommand = DeleteStageCommand;
exports.DeleteStorageConfigurationCommand = DeleteStorageConfigurationCommand;
exports.DisconnectParticipantCommand = DisconnectParticipantCommand;
exports.EventErrorCode = EventErrorCode;
exports.GetCompositionCommand = GetCompositionCommand;
exports.GetEncoderConfigurationCommand = GetEncoderConfigurationCommand;
exports.GetIngestConfigurationCommand = GetIngestConfigurationCommand;
exports.GetParticipantCommand = GetParticipantCommand;
exports.GetPublicKeyCommand = GetPublicKeyCommand;
exports.GetStageCommand = GetStageCommand;
exports.GetStageSessionCommand = GetStageSessionCommand;
exports.GetStorageConfigurationCommand = GetStorageConfigurationCommand;
exports.IVSRealTime = IVSRealTime;
exports.IVSRealTimeClient = IVSRealTimeClient;
exports.IVSRealTimeServiceException = IVSRealTimeServiceException$1;
exports.ImportPublicKeyCommand = ImportPublicKeyCommand;
exports.IngestProtocol = IngestProtocol;
exports.InternalServerException = InternalServerException$1;
exports.ListCompositionsCommand = ListCompositionsCommand;
exports.ListEncoderConfigurationsCommand = ListEncoderConfigurationsCommand;
exports.ListIngestConfigurationsCommand = ListIngestConfigurationsCommand;
exports.ListParticipantEventsCommand = ListParticipantEventsCommand;
exports.ListParticipantReplicasCommand = ListParticipantReplicasCommand;
exports.ListParticipantsCommand = ListParticipantsCommand;
exports.ListPublicKeysCommand = ListPublicKeysCommand;
exports.ListStageSessionsCommand = ListStageSessionsCommand;
exports.ListStagesCommand = ListStagesCommand;
exports.ListStorageConfigurationsCommand = ListStorageConfigurationsCommand;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ParticipantProtocol = ParticipantProtocol;
exports.ParticipantRecordingMediaType = ParticipantRecordingMediaType;
exports.PendingVerification = PendingVerification$1;
exports.PipBehavior = PipBehavior;
exports.PipPosition = PipPosition;
exports.ResourceNotFoundException = ResourceNotFoundException$1;
exports.ServiceQuotaExceededException = ServiceQuotaExceededException$1;
exports.StartCompositionCommand = StartCompositionCommand;
exports.StartParticipantReplicationCommand = StartParticipantReplicationCommand;
exports.StopCompositionCommand = StopCompositionCommand;
exports.StopParticipantReplicationCommand = StopParticipantReplicationCommand;
exports.TagResourceCommand = TagResourceCommand;
exports.ThumbnailRecordingMode = ThumbnailRecordingMode;
exports.ThumbnailStorageType = ThumbnailStorageType;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UpdateIngestConfigurationCommand = UpdateIngestConfigurationCommand;
exports.UpdateStageCommand = UpdateStageCommand;
exports.ValidationException = ValidationException$1;
exports.VideoAspectRatio = VideoAspectRatio;
exports.VideoFillMode = VideoFillMode;
exports.paginateListCompositions = paginateListCompositions;
exports.paginateListEncoderConfigurations = paginateListEncoderConfigurations;
exports.paginateListIngestConfigurations = paginateListIngestConfigurations;
exports.paginateListParticipantEvents = paginateListParticipantEvents;
exports.paginateListParticipantReplicas = paginateListParticipantReplicas;
exports.paginateListParticipants = paginateListParticipants;
exports.paginateListPublicKeys = paginateListPublicKeys;
exports.paginateListStageSessions = paginateListStageSessions;
exports.paginateListStages = paginateListStages;
exports.paginateListStorageConfigurations = paginateListStorageConfigurations;
