// @aws-sdk/client-ivs-realtime/dist-es/models/enums.js
export const IngestProtocol = {
    RTMP: "RTMP",
    RTMPS: "RTMPS",
};
export const ParticipantRecordingMediaType = {
    AUDIO_ONLY: "AUDIO_ONLY",
    AUDIO_VIDEO: "AUDIO_VIDEO",
    NONE: "NONE",
};
export const ThumbnailRecordingMode = {
    DISABLED: "DISABLED",
    INTERVAL: "INTERVAL",
};
export const ThumbnailStorageType = {
    LATEST: "LATEST",
    SEQUENTIAL: "SEQUENTIAL",
};
export const VideoAspectRatio = {
    AUTO: "AUTO",
    PORTRAIT: "PORTRAIT",
    SQUARE: "SQUARE",
    VIDEO: "VIDEO",
};
export const VideoFillMode = {
    CONTAIN: "CONTAIN",
    COVER: "COVER",
    FILL: "FILL",
};
export const PipBehavior = {
    DYNAMIC: "DYNAMIC",
    STATIC: "STATIC",
};
export const PipPosition = {
    BOTTOM_LEFT: "BOTTOM_LEFT",
    BOTTOM_RIGHT: "BOTTOM_RIGHT",
    TOP_LEFT: "TOP_LEFT",
    TOP_RIGHT: "TOP_RIGHT",
};
export const ParticipantProtocol = {
    RTMP: "RTMP",
    RTMPS: "RTMPS",
    UNKNOWN: "UNKNOWN",
    WHIP: "WHIP",
};
export const EventErrorCode = {
    BITRATE_EXCEEDED: "BITRATE_EXCEEDED",
    B_FRAME_PRESENT: "B_FRAME_PRESENT",
    INSUFFICIENT_CAPABILITIES: "INSUFFICIENT_CAPABILITIES",
    INTERNAL_SERVER_EXCEPTION: "INTERNAL_SERVER_EXCEPTION",
    INVALID_AUDIO_CODEC: "INVALID_AUDIO_CODEC",
    INVALID_INPUT: "INVALID_INPUT",
    INVALID_PROTOCOL: "INVALID_PROTOCOL",
    INVALID_STREAM_KEY: "INVALID_STREAM_KEY",
    INVALID_VIDEO_CODEC: "INVALID_VIDEO_CODEC",
    PUBLISHER_NOT_FOUND: "PUBLISHER_NOT_FOUND",
    QUOTA_EXCEEDED: "QUOTA_EXCEEDED",
    RESOLUTION_EXCEEDED: "RESOLUTION_EXCEEDED",
    REUSE_OF_STREAM_KEY: "REUSE_OF_STREAM_KEY",
    STREAM_DURATION_EXCEEDED: "STREAM_DURATION_EXCEEDED",
};
